validator/lib/jsonapi/validator/resource.rb

```diff
@@ -0,0 +1,123 @@
+require 'jsonapi/validator/document'
+require 'jsonapi/validator/relationship'
+
+module JSONAPI
+  module Validator
+    class Resource
```

Comment:
Yeah I think we agree on all points, actually.

Regarding 5 - just spitballing here, but I could see it as a low-level tool for spec helpers that need to traverse and validate sections of a jsonapi document. My [jsonapi_spec_helpers](https://jsonapi-suite.github.io/jsonapi_spec_helpers/) gem could use it.

Regarding the separate validator gem - no disagreement, but curious on the overall direction/usage you expect of it. Current alternatives like [sinatra-param](https://github.com/zendesk/stronger_parameters) and [strong_parameters](https://github.com/zendesk/stronger_parameters) no good?